### DISCUSSION TITLE
Infinite canvas and `mousemove` event

### DISCUSSION THREAD

**Discussion by bgrand-ch** [GrapesJS/grapesjs#6097] [Show and tell]
Hello,

I use the infinite canvas and the "grapesjs-click" plugin.

The `mousemove` event in the "grapesjs-click" plugin is used to make an HTML element follow the mouse cursor.

The X and Y positions of the HTML element were transformed when the cursor passed over the iframe (`.gjs-frame`).

Below is the code of the "grapesjs-click" plugin to retrieve the coordinates of the mouse cursor with the infinite canvas:

```ts
function getMousePosition (editor: Editor, event: MouseEvent) {
  if (!event.isTrusted) {
    return editor.Canvas.getMouseRelativeCanvas(event, {})
  }

  return {
    x: event.clientX,
    y: event.clientY
  }
}
```

More info:
https://github.com/bgrand-ch/grapesjs-click/commit/68bf26cdeafcb718c361be67d37ae74fe6e485e3

**Comment by artf** [comment]
Nice 👏🚀
